analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ParameterLoaderTest.cs

```diff
@@ -81,28 +81,32 @@ public void SetParameterValues_WhenGivenSonarLintFileHasIntParameterType_Populat
         public void SetParameterValues_WhenGivenSonarLintFileHasStringParameterType_OnlyOneParameter_PopulatesProperty()
         {
             // Arrange
-            var compilation = CreateCompilationWithOption("ResourceTests\\SonarLintXml\\RuleWithStringParameter\\SonarLint.xml");
+            var parameterValue = "1";
+            var filePath = GenerateSonarLintWithParametrizedRule("S2342", "flagsAttributeFormat", parameterValue);
+            var compilation = CreateCompilationWithOption(filePath);
             var analyzer = new EnumNameShouldFollowRegex(); // Cannot use mock because we use reflection to find properties.
 
             // Act
             ParameterLoader.SetParameterValues(analyzer, compilation.SonarLintFile());
 
             // Assert
-            analyzer.FlagsEnumNamePattern.Should().Be("1"); // value from XML file
+            analyzer.FlagsEnumNamePattern.Should().Be(parameterValue); // value from XML file
         }
 
         [TestMethod]
         public void SetParameterValues_WhenGivenSonarLintFileHasBooleanParameterType_OnlyOneParameter_PopulatesProperty()
         {
             // Arrange
-            var compilation = CreateCompilationWithOption("ResourceTests\\SonarLintXml\\RuleWithBooleanParameter\\SonarLint.xml");
+            var parameterValue = true;
```

Comment:
The point of this is to be clearer about what you are actually comparing to, right?
I'd put it over the assertion thought.

Reply:
Yes exactly. So the other option would be just to hardcode in both places the value.
```cs
var filePath = GenerateSonarLintXmlWithParametrizedRule("S1451", "isRegularExpression", "true");
[...]
analyzer.IsRegularExpression.Should().Be("true");
```
but I think having a single variable to reference makes sure that there is no mismatch between the input and what we expect. Does it make sense to you?